src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -78,4 +85,35 @@ public static Map<String, String> parseStringToStringMap(XContentParser parser)
         return map;
     }
 
+    /**
+     * Parse content parser to {@link java.time.Instant}.
+     *
+     * @param parser json based content parser
+     * @return instance of {@link java.time.Instant}
+     * @throws IOException IOException if content can't be parsed correctly
+     */
+    public static Instant parseInstant(XContentParser parser) throws IOException {
+        if (parser.currentToken() == null || parser.currentToken() == XContentParser.Token.VALUE_NULL) {
+            return null;
+        }
+        if (parser.currentToken().isValue()) {
+            return Instant.ofEpochMilli(parser.longValue());
+        }
+        return null;
+    }
+
+    /**
+     * Generates a user string formed by the username, backend roles, roles and requested tenants separated by '|'
+     * (e.g., john||own_index,testrole|__user__, no backend role so you see two verticle line after john.).
+     * This is the user string format used internally in the OPENSEARCH_SECURITY_USER_INFO_THREAD_CONTEXT and may be
+     * parsed using User.parse(string).
+     * @param client Client containing user info. A public API request will fill in the user info in the thread context.
+     * @return parsed user object
+     */
+    public static User getUserContext(Client client) {
+        String userStr = client.threadPool().getThreadContext().getTransient(ConfigConstants.OPENSEARCH_SECURITY_USER_INFO_THREAD_CONTEXT);
+        logger.debug("Filtering result by " + userStr);
```

Comment:
Do we want to log user info here even in debug mode?

Reply:
I think this is okay to have for debugging, ML commons and AD plugin have the same exact line